src/server/scripts/Kalimdor/HallsOfOrigination/boss_isiset.cpp

```diff
@@ -0,0 +1,1071 @@
+                    events.ScheduleEvent(EVENT_SUPERNOVA, urand(45,50)*IN_MILLISECONDS);
+                    break;
+                case EVENT_IMAGES_ATTACK:
+                {
+                    DummyEntryCheckPredicate pred;
+                    summons.DoAction(ACTION_IMAGES_SET_AGGRESSIVE, pred);
+                    break;
+                }
+                case EVENT_DESPAWN_IMAGES:
+                {
+                    DummyEntryCheckPredicate pred;
+                    summons.DoAction(ACTION_IMAGES_DESPAWN, pred);
+                    instance->SetData(DATA_ISISET_PHASE, instance->GetData(DATA_ISISET_PHASE) + 1);
+                    me->RemoveAurasDueToSpell(SPELL_ASTRAL_SHIFT);
+                    RescheduleEvents();
+                    break;
+                }
+                default:
+                    break;
+                }
+            }
+
+            DoMeleeAttackIfReady();
+        }
+
+    private:
+        void RescheduleEvents()
+        {
+            events.Reset();
+            events.ScheduleEvent(EVENT_START_ATTACK, 1200);
+            if (instance->GetData(DATA_ISISET_ASTRAL_RAIN_ALIVE) == 1)
+                events.ScheduleEvent(EVENT_ASTRAL_RAIN, urand(4, 6)*IN_MILLISECONDS);
+            if (instance->GetData(DATA_ISISET_CELESTIAL_CALL_ALIVE) == 1)
+                events.ScheduleEvent(EVENT_ASTRAL_FAMILIAR, 7200);
+            if (instance->GetData(DATA_ISISET_VEIL_OF_SKY_ALIVE) == 1)
+                events.ScheduleEvent(EVENT_VEIL_OF_SKY, urand(8, 10)*IN_MILLISECONDS);
+            events.ScheduleEvent(EVENT_SUPERNOVA, urand(12, 17)*IN_MILLISECONDS);
+        }
+
+        // Mirror Image transition phase
+        void MirrorImage()
+        {
+            events.Reset();
+            instance->SendEncounterUnit(ENCOUNTER_FRAME_UPDATE_PRIORITY, me, 0);
+            mirrorImageDied = false;
+
+            DoCast(SPELL_ASTRAL_SHIFT);
+            DoCast(SPELL_MIRROR_IMAGE_STARRY_SKY_SPAWNER);
+            DoCast(SPELL_MIRROR_IMAGE_SPAWNER);
+
+            events.ScheduleEvent(EVENT_IMAGES_ATTACK, 2000);
+        }
+
+        bool mirrorImageDied;
+    };
+
+    CreatureAI* GetAI(Creature* creature) const override
+    {
+        return GetHallsOfOriginationAI<boss_isisetAI>(creature);
+    }
+};
+
+// 39795 - Celestial Familiar
+class npc_celestial_familiar : public CreatureScript
+{
+public:
+    npc_celestial_familiar() : CreatureScript("npc_celestial_familiar") { }
+
+    struct npc_celestial_familiarAI : public ScriptedAI
+    {
+        npc_celestial_familiarAI(Creature* creature) : ScriptedAI(creature)
+        {
+            me->SetReactState(REACT_PASSIVE);
+            events.ScheduleEvent(EVENT_ORB_ADD_VISUAL, 1200);
+            events.ScheduleEvent(EVENT_ORB_SET_AGGRESSIVE, 2400);
+            events.ScheduleEvent(EVENT_ORB_ARCANE_BARRAGE, urand(4,6)*IN_MILLISECONDS);
+        }
+
+        void UpdateAI(uint32 diff) override
+        {
+            events.Update(diff);
```

Comment:
No victim check? creature will never enter in evade :/

Reply:
It needs to execute some events while reacting passive, I couldn't remember any other way :/

Reply:
It's a boss summon and despawns anyway

Reply:
Enters combat rightaway now, so UpdateVictim check is back.